_signal_processing/krakenSDR_signal_processor.py

```diff
@@ -239,6 +241,46 @@ def resetPeakHold(self):
         if self.spectrum_fig_type == "Single":
             self.peak_hold_spectrum = np.ones(self.spectrum_window_size) * -200
 
+    def calculate_squelch(self, sampling_freq, N):
+        def is_enabled_auto_squelch(v):
+            return v == "Auto" or (v == "Default" and self.vfo_default_squelch_mode == "Auto")
+
+        def find_nearest(array, value):
+            array = np.asarray(array)
+            idx = (np.abs(array - value)).argmin()
+            return idx, array[idx]
+
+        if self.spectrum_fig_type == "Single":
+            spectrum_index = 1
+        else:
+            spectrum_index = 2
+        sensor1_spec = self.spectrum[spectrum_index, :]
+        default_auto_db_offset = 5
+        default_auto_channel_db_offset = 3
+
+        auto_squelch = any(is_enabled_auto_squelch(vfo_squelch_mode) for vfo_squelch_mode in self.vfo_squelch_mode)
+        if auto_squelch:
+            sensor1_spec_mean = np.mean(sensor1_spec)
+            vfo_auto_squelch = sensor1_spec_mean + default_auto_db_offset
+            self.vfo_squelch = [vfo_auto_squelch] * self.max_vfos
+
+        moving_avg_freq_window = 100_000  # 100kHz
```

Comment:
@redradist Please move this to `variables`

Reply:
See above answer